service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -177,20 +185,36 @@ public ResponseEntity<PostMethodResponse> postMethod(PostMethodRequest postMetho
       UUID methodId = UUID.randomUUID();
       UUID runSetId = UUID.randomUUID();
 
+      GithubUrlDetailsManager.GithubUrlComponents githubUrlComponents =
+          githubUrlDetailsManager.extractDetailsFromUrl(rawMethodUrl);
```

Comment:
What happens if the URL is not a github url?

Reply:
I wonder if this should just be a `UrlDetailsManager`, even if its purpose for now is to extract GitHub details. It could also handle the question "is this a GitHub URL?"

Reply:
Since we only support github urls, I didn't find it necessary to name this something more broad. In the future, other urls will most likely follow a different format, so we'll need an additional method in the manager to handle these specific types.